### ISSUE TITLE
Increase test coverage for at_app pipeline

### ISSUE THREAD

**Issue by xavierchanth** [atsign-foundation/at_app_cli#68]
Lead: @XavierChanth 

There are many points of failure that could cause at_app to fail when generating the app.

There is also a lot of code to cover within at_app, the template bundlers, template generators, and generated templates all need to be covered eventually. As we add more samples to at_app, this will quickly become harder to maintain.

My goal is to increase test coverage to handle as much of this as possible.

**Preparation Work**
- [x] ~~Custom docker image for actions~~ (replaced with reusable actions)
- [ ] Custom dependabot
  - [ ] Regularly check for the latest kotlin/gradle versions - not sure if this is possible
  - [ ] Update them in at_app_create or notify the devs via actions
 - [x] Investigate offloading some testing away from actions (mainly generated app testing) - determined to be unnecessary work
 - [x] Reusable composite actions for setup

**Unit/Integration Tests** - Focus for PR 35
- [ ] at_app
- [x] at_app_flutter
- [ ] at_app_create
- [ ] at_app_bundler

**Functional Tests**
- [x] `melos run build` #87 
- [x] then verify each of the templates #111  
  - [x] app
  - [x] snackbar
  - [x] snackbar_sender
  - [x] at_contacts_flutter
  - [x] at_onboarding_flutter
  - [x] chit_chat

**Comment by xavierchanth** [comment]
In the current sprint, I would like to at least start by determining the approach that I will use to tackle this task.

**Comment by xavierchanth** [comment]
Going to postpone this issue until after #77 and possibly until after at_template is fully released.
More of the functionality may be moved from at_app to at_template.

**Comment by xavierchanth** [comment]
Picking this back up where I left off in PR35, still some decisions to make about how to make this project scalable and maintainable.

**Comment by xavierchanth** [comment]
We changed this from 3 to 5 due to my capacity during sprint planning. I've opted to backlog #105 to focus more on this issue. There isn't much I can progress with on #105 at the moment. Bumping this from 3 back to 5 sp.

**Comment by xavierchanth** [comment]
I had an issue with the CI failing on PR which should now be resolved [here](https://github.com/atsign-foundation/at_app/pull/139)

**Comment by xavierchanth** [comment]
I will await the next stable version of mason, and #105 before continuing on this issue. Any work I do now may become irrelevant in the coming months.

**Comment by ksanty** [comment]
@XavierChanth has anything changed the status with this card?

**Comment by ksanty** [comment]
@XavierChanth has anything changed on status with this card?

**Comment by xavierchanth** [comment]
> @XavierChanth has anything changed the status with this card?

Nothing recently, at_app is stable at the moment, so there hasn't been a rush to work on this.

**Comment by xavierchanth** [comment]
Closing this, since this issue is stale and at_app is stable.
